src/connector/src/source/iceberg/mod.rs

```diff
@@ -157,16 +157,70 @@ impl SplitEnumerator for IcebergSplitEnumerator {
     }
 }
 
+pub enum IcebergTimeTravelInfo {
+    Version(i64),
+    TimeStampMs(i64),
+}
+
 impl IcebergSplitEnumerator {
     pub async fn list_splits_batch(
         &self,
+        time_traval_info: Option<IcebergTimeTravelInfo>,
         batch_parallelism: usize,
     ) -> ConnectorResult<Vec<IcebergSplit>> {
         if batch_parallelism == 0 {
             bail!("Batch parallelism is 0. Cannot split the iceberg files.");
         }
         let table = self.config.load_table().await?;
-        let snapshot_id = table.current_table_metadata().current_snapshot_id.unwrap();
+        let snapshot_id = match time_traval_info {
+            Some(IcebergTimeTravelInfo::Version(version)) => {
+                match &table.current_table_metadata().snapshots {
```

Comment:
Seems there is an interface to get the specific snapshot version from TableMetadata. https://github.com/icelake-io/icelake/blob/54fd72fbd1dd8c592f05eeeb79223c8a6a33c297/icelake/src/types/in_memory.rs#L2586C12-L2586C20

Reply:
+1